### ISSUE TITLE
Reacting to onSubmit Promise rejection

### ISSUE THREAD

**Issue by taras-yurts** [final-form/final-form#464]
### Submitting a **bug report** or a **feature request**:

Technically a feature request based on the contradiction with docs wording

### Current behavior:

Currently, the form state doesn't change if there is some Promise rejection in the onSubmit handler, moreover `submitSucceeded` property turns true if that happens. Even though, according to [the docs](https://final-form.org/docs/final-form/types/Config#3-asynchronous-with-a-promise), there is wording about 'leaving rejection for when there is a server or communications error' and it seems that the form should consider some handling for rejection(server or communications error) case.

### Expected behavior:

From [the docs](https://final-form.org/docs/final-form/types/Config#3-asynchronous-with-a-promise) it seems that the form's submission shouldn't be considered as succeeded in case there is some rejection in the submit handler or at least there should be some forms property or handler reacting to the rejection.

### Environment:

react-final-form: 6.5.9
final-form: 4.20.7
node: 16.14.2
npm: 8.1.4

OS: Windows
Browser: Chrome 106.0.5249.91

**Comment by andriy-kudrya-eleks** [comment]
Yeah, setting ```submitSucceeded = true``` after request (and promise) is a surprising behavior. Is it designed to be this way or is it a bug?

I'd rather expect setting ```submitFailed = ture```. In a case ```submitFailed``` is reserved only for a case when request promise is resolved with error object then it is better to not to change form state at all. Currently, final-form behaves as if request was successful.
